plugins/woocommerce/includes/abstracts/abstract-wc-product.php

```diff
@@ -148,7 +148,7 @@ public function __construct( $product = 0 ) {
 	 * @return string
 	 */
 	public function get_type() {
-		return isset( $this->product_type ) ? $this->product_type : 'simple';
+		return $this->product_type ?? 'simple';
```

Comment:
Most of the failing tests have the same error:

```
Unexpected incorrect usage notice for product_type
Failed asserting that an array is empty.
```

The easiest path to solving this is to roll this change back:

```suggestion
		return isset( $this->product_type ) ? $this->product_type : 'simple';
```

Context: see the dynamic property usage in [`WC_Abstract_Legacy_Product::__get()`](https://github.com/woocommerce/woocommerce/blob/6.6.1/plugins/woocommerce/includes/legacy/abstract-wc-legacy-product.php#L66).